src/js/msp.js

```diff
@@ -72,9 +82,37 @@ const MSP = {
 
         for (const chunk of data) {
             switch (this.state) {
+            case this.decoder_states.CLI_COMMAND:
+                switch (chunk) {
+                    case this.symbols.END_OF_TEXT:
+                        this.cli_output.push(this.cli_buffer.join(''));
+                        this.cli_buffer.length = 0;
```

Comment:
Should we empty the array instead ?

Reply:
Any reference to the array remains if you "set" the array using ` = [];`

Setting length to 0 means any reference to the array is also cleared. In this case the buffer should be cleared, and no reference should remain.

https://stackoverflow.com/questions/1232040/how-do-i-empty-an-array-in-javascript